solidity/contracts/contracts/HashingTest.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: Apache-2.0
+
+pragma solidity ^0.8.0;
+
+contract HashingTest {
+    bytes32 public lastCheckpoint;
+    address[] public state_validators;
+    uint256[] public state_powers;
+    uint256 public state_nonce;
+
+    function IterativeHash(
+        address[] memory _validators,
+        uint256[] memory _powers,
+        uint256 _valsetNonce,
+        bytes32 _peggyId
+    ) public {
+        // bytes32 encoding of the string "checkpoint"
+        bytes32 methodName = 0x636865636b706f696e7400000000000000000000000000000000000000000000;
+
+        bytes32 checkpoint = keccak256(abi.encode(_peggyId, methodName, _valsetNonce));
+
+        // Iterative hashing of valset
+        {
+            for (uint256 i = 0; i < _validators.length; i = i + 1) {
+                // Check that validator powers are decreasing or equal (this allows the next
+                // caller to break out of signature evaluation ASAP to save more gas)
+                if (i != 0) {
+                    require(
+                        !(_powers[i] > _powers[i - 1]),
+                        "Validator power must not be higher than previous validator in batch"
+                    );
+                }
+                checkpoint = keccak256(abi.encode(checkpoint, _validators[i], _powers[i]));
+            }
+        }
+
```

Comment:
This function will return without error even if fed an empty validator set - is that intended?

Reply:
Sure, don't see why not.